src/coreclr/jit/importercalls.cpp

```diff
@@ -8139,6 +8140,13 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                             result = NI_System_Span_get_Length;
                         }
                     }
+                    else if (strcmp(className, "SpanHelpers") == 0)
+                    {
+                        if (strcmp(methodName, "SequenceEqual") == 0)
```

Comment:
I know this is completely unrelated to the fix or changes being done, sorry... but has anyone ever tried reversing the `methodName` and `className` tests for a performance hack in the JIT itself?

Reply:
@IDisposable `lookupNamedIntrinsics` never show up in our JIT traces so we don't bother. This code is only executed for methods with `[Intrinsic]` attribute so for 99% of methods it doesn't kick in.

We could use here a Trie/binary search if it was a real problem